mabwiser/treebandit.py

```diff
@@ -0,0 +1,163 @@
+# -*- coding: utf-8 -*-
+# SPDX-License-Identifier: Apache-2.0
+
+from copy import deepcopy
+from typing import Union, Dict, List, NoReturn, Optional, Callable
+
+import numpy as np
+from sklearn.tree import DecisionTreeClassifier
+
+from mabwiser.base_mab import BaseMAB
+from mabwiser.greedy import _EpsilonGreedy
+from mabwiser.linear import _Linear
+from mabwiser.popularity import _Popularity
+from mabwiser.rand import _Random
+from mabwiser.softmax import _Softmax
+from mabwiser.thompson import _ThompsonSampling
+from mabwiser.ucb import _UCB1
+from mabwiser.utils import reset, argmax, Arm, Num, _BaseRNG
+
+
+class _TreeBandit(BaseMAB):
+    def __init__(self, rng: _BaseRNG, arms: List[Arm], n_jobs: int, backend: Optional[str],
+                 lp: Union[_EpsilonGreedy, _Linear, _Popularity, _Random, _Softmax, _ThompsonSampling, _UCB1],
+                 tree_parameters: Dict):
+        super().__init__(rng, arms, n_jobs, backend)
+        self.lp = lp
+        self.tree_parameters = tree_parameters
+        self.tree_parameters["random_state"] = rng.seed
+
+        self.arm_to_tree = None
+        self.arm_to_rewards = None
+
+        # Initialize the arm expectations to nan
+        # When there are neighbors, expectations of the underlying learning policy is used
```

Comment:
not sure what does this comment mean?

Reply:
I put that comment and initialization to be consistent with the other neighborhood policies. What it suggests is to initialize arm_to_expectation values to nan for the nhpolicy and we'll use the arm_to_expectation from the learning policy.